simple_kp/engine.py

```diff
@@ -2,6 +2,8 @@
 from collections import defaultdict
 import logging
 import os
+import re
+import itertools
```

Comment:
I like to sort standard imports alphabetically, like [isort](https://pycqa.github.io/isort/).